jbehave-core/src/main/java/org/jbehave/core/embedder/PerformableTree.java

```diff
@@ -1171,19 +1219,29 @@ protected void performRestartableSteps(RunContext context)
 		}
 
         protected void perform(RunContext context, GivenStories stories) throws InterruptedException {
+            perform(context, stories, Optional.empty());
+        }
+
+        void perform(RunContext context, GivenStories stories, Optional<UUIDExceptionWrapper> causeToSkip)
+                throws InterruptedException {
             perform(beforeSteps, context, r -> r.beforeScenarioSteps(Stage.BEFORE),
-                r -> r.afterScenarioSteps(Stage.BEFORE));
-            performGivenStories(context, givenStories, stories);
-            performRestartableSteps(context);
+                r -> r.afterScenarioSteps(Stage.BEFORE), Optional.empty());
```

Comment:
Before and after scenario steps not marked with causeToSkip

Reply:
what happens with After scenario steps declared in Lifecycle?